code/src/containers/Survey/Text/Text.js

```diff
@@ -0,0 +1,24 @@
+import React, { useState } from 'react'
+
```

Comment:
really nice that you extracted the inputs to reuse them. Although, should allergens actually be required? Maybe you could pass required as a prop instead? To make it even more modular?